src/Components/App/App.js

```diff
@@ -12,41 +11,42 @@ class App extends Component {
     super()
     this.state = {
       movies: [],
-      clicked: false
+      clicked: false,
+      singleMovieDetails: [],
+      err: ''
     }
   }
 
   componentDidMount() {
     fetch('https://rancid-tomatillos.herokuapp.com/api/v2/movies')
       .then(response => response.json())
       .then(data => this.setState({ movies: data.movies, clicked: false }))
-      // .then(err => this.setState({ err: err.message }))
-  }
-
-  componentWillUnmount() {
-
+      .catch(() => {this.setState({err: 'Refresh Page'})})
   }
 
   viewMovie = (id) => {
     console.log('CHECKING', this.state.movies)
     const findMovie = this.state.movies.find(movie => movie.id === id)
-    this.setState({ movies: [findMovie], clicked: true})
+    fetch(`https://rancid-tomatillos.herokuapp.com/api/v2/movies/${findMovie.id}`)
+    .then(response => response.json())
+    .then(data => this.setState({clicked: true, singleMovieDetails: data.movie}))
```

Comment:
Glad we figured out to add the fetch within our App file, lets remember to always change the state within the app component, because we spiraled and made it more complicated than it needed to be!
- code loookin cleaaaan!